pkg/client/testing/core/fixture.go

```diff
@@ -286,6 +297,15 @@ func (t *tracker) add(obj runtime.Object, replaceExisting bool) error {
 			return err
 		}
 
+		// Propagate namespace to the new object if hasn't already been set.
+		if len(newMeta.GetNamespace()) == 0 {
+			newMeta.SetNamespace(ns)
+		}
+
+		if newMeta.GetNamespace() != ns {
```

Comment:
@smarterclayton but this mimics the behavior you have when invoking similar operations against regular cluster. Operations spanning multiple namespaces does not include `Create` or `Update`, these are always targeted against specific namespace, and this is what `add` doe here. I'd just update this to include the namespace names, like this:
```
the namespace from the provided object <newMeta.GetNamespace()> does not match the namespace <ns>
```